streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -806,6 +803,20 @@ private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
         rocksDBMetricsRecordingService = maybeCreateRocksDBMetricsRecordingService(clientId, config);
     }
 
+    private long getCacheSizePerThread(final int numStreamThreads) {
+        return totalCacheSize / (numStreamThreads + ((globalTaskTopology != null) ? 1 : 0));
+    }
+
+    private void resizeThreadCache(final int numStreamThreads) {
+        if (numStreamThreads < 0) {
```

Comment:
Can it be smaller than `0` ? Should the test be `<= 0` or `< 1` instead?

Reply:
It can be zero if you have a global thread, but since this is internal the check might not be entirely necessary

Reply:
Yes, it can be zero, but the check say `< 0`, so it would always evaluate to false?

And if we have zero threads, we should not resize the cache as we might end up in an infinite loop? But we would only call this method if we "shrink", ie, if the thread count grows, but it can never grow from negative to zero, right?

Reply:
That is a good point. Maybe what we need to do it put a minimum size of cache to limit how many stream threads an instance can have?

Reply:
Well, `getCacheSizePerThread` would eventually return zero, what means that every put() into the cache would result in an immediate eviction. So I don't think we need to do anything for this corner case).

Reply:
that is a good point